src/module/item/affliction/document.ts

```diff
@@ -65,6 +71,22 @@ class AfflictionPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extend
         await this.update({ system: { stage } });
     }
 
+    get onsetDuration(): number {
+        const value = this.totalDuration(this.system.onset);
+        return value === Infinity ? 0 : value;
+    }
+
+    totalDuration(duration: AfflictionDuration | AfflictionOnset | undefined): number {
```

Comment:
This needs to be a getter and match effect's signature as best as possible.

Reply:
in this case we will have 2 similar getters, 1 for duration of affliction(max duration) and 1 for stage duration

Reply:
That sounds fine to me. They're different concepts, and totalDuration is more or less the same as the effect one.